<a href="https://colab.research.google.com/github/sousadevelop/Data-Science-Studies/blob/main/tratamento_enem_2019.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [9]:
import numpy as np
import pandas as pd

In [10]:
dados = pd.read_csv('microdados_enem_2019_sp.csv', sep=';', encoding='iso-8859-1')

In [11]:
dados.head()

NU_INSCRICAO  CO_MUNICIPIO_RESIDENCIA NO_MUNICIPIO_RESIDENCIA  \
0     1,90E+11                  3546801            Santa Isabel   
1     1,90E+11                  3538204             Pinhalzinho   
2     1,90E+11                  3550308               São Paulo   
3     1,90E+11                  3548708   São Bernardo do Campo   
4     1,90E+11                  3549409    São Joaquim da Barra   

   CO_UF_RESIDENCIA SG_UF_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0                35               SP        22       M                1   
1                35               SP        19       M                1   
2                35               SP        17       F                1   
3                35               SP        19       M                1   
4                35               SP        37       M                2   

   TP_COR_RACA  TP_NACIONALIDADE  ...  NU_NOTA_MT TP_LINGUA  \
0            3                 1  ...      6002.0         0   
1            3                 1  ...         NaN         1   
2            3                 1  ...      7318.0         0   
3            3                 1  ...      3932.0         1   
4            1                 1  ...         NaN         1   

   TP_STATUS_REDACAO NU_NOTA_COMP1  NU_NOTA_COMP2  NU_NOTA_COMP3  \
0                1.0         160.0          200.0          180.0   
1                NaN           NaN            NaN            NaN   
2                1.0         160.0          120.0          200.0   
3                1.0          80.0          160.0          120.0   
4                NaN           NaN            NaN            NaN   

   NU_NOTA_COMP4  NU_NOTA_COMP5  NU_NOTA_REDACAO  Q025  
0          200.0          200.0            940.0     B  
1            NaN            NaN              NaN     A  
2          200.0          200.0            880.0     A  
3          100.0          100.0            560.0     B  
4            NaN            NaN              NaN     B  

[5 rows x 44 columns]

In [12]:
dados.shape

(813772, 44)

In [13]:
dados.dtypes

NU_INSCRICAO                object
CO_MUNICIPIO_RESIDENCIA      int64
NO_MUNICIPIO_RESIDENCIA     object
CO_UF_RESIDENCIA             int64
SG_UF_RESIDENCIA            object
NU_IDADE                     int64
TP_SEXO                     object
TP_ESTADO_CIVIL              int64
TP_COR_RACA                  int64
TP_NACIONALIDADE             int64
CO_MUNICIPIO_NASCIMENTO    float64
NO_MUNICIPIO_NASCIMENTO     object
CO_UF_NASCIMENTO           float64
SG_UF_NASCIMENTO            object
TP_ST_CONCLUSAO              int64
TP_ANO_CONCLUIU              int64
TP_ESCOLA                    int64
TP_ENSINO                  float64
IN_TREINEIRO                 int64
CO_ESCOLA                  float64
CO_MUNICIPIO_ESC           float64
NO_MUNICIPIO_ESC            object
CO_UF_ESC                  float64
SG_UF_ESC                   object
TP_DEPENDENCIA_ADM_ESC     float64
TP_LOCALIZACAO_ESC         float64
TP_SIT_FUNC_ESC            float64
TP_PRESENCA_CN               int64
TP_PRESENCA_CH      

## Apagando colunas

In [14]:
dados1 = dados.drop(columns=['CO_MUNICIPIO_RESIDENCIA', 'CO_UF_RESIDENCIA','SG_UF_RESIDENCIA','CO_MUNICIPIO_NASCIMENTO', 'NO_MUNICIPIO_NASCIMENTO','CO_UF_NASCIMENTO','SG_UF_NASCIMENTO',
                              'TP_ANO_CONCLUIU','TP_ENSINO','CO_MUNICIPIO_ESC','CO_UF_ESC','SG_UF_ESC', 'TP_DEPENDENCIA_ADM_ESC','TP_LOCALIZACAO_ESC','TP_SIT_FUNC_ESC'], )

In [15]:
dados1.head(8)

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0     1,90E+11            Santa Isabel        22       M                1   
1     1,90E+11             Pinhalzinho        19       M                1   
2     1,90E+11               São Paulo        17       F                1   
3     1,90E+11   São Bernardo do Campo        19       M                1   
4     1,90E+11    São Joaquim da Barra        37       M                2   
5     1,90E+11              Embu-Guaçu        17       F                1   
6     1,90E+11     São José dos Campos        28       F                1   
7     1,90E+11     São José dos Campos        25       F                2   

   TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ESCOLA  IN_TREINEIRO  \
0            3                 1                1          1             0   
1            3                 1                1          1             0   
2            3                 1                2          3             0   
3            3                 1                2          2             0   
4            1                 1                1          1             0   
5            3                 1                2          2             0   
6            1                 1                1          1             0   
7            1                 1                1          1             0   

   ...  NU_NOTA_MT TP_LINGUA  TP_STATUS_REDACAO  NU_NOTA_COMP1  NU_NOTA_COMP2  \
0  ...      6002.0         0                1.0          160.0          200.0   
1  ...         NaN         1                NaN            NaN            NaN   
2  ...      7318.0         0                1.0          160.0          120.0   
3  ...      3932.0         1                1.0           80.0          160.0   
4  ...         NaN         1                NaN            NaN            NaN   
5  ...      4192.0         0                1.0          120.0          120.0   
6  ...         NaN         0                NaN            NaN            NaN   
7  ...         NaN         1                NaN            NaN            NaN   

   NU_NOTA_COMP3  NU_NOTA_COMP4  NU_NOTA_COMP5  NU_NOTA_REDACAO  Q025  
0          180.0          200.0          200.0            940.0     B  
1            NaN            NaN            NaN              NaN     A  
2          200.0          200.0          200.0            880.0     A  
3          120.0          100.0          100.0            560.0     B  
4            NaN            NaN            NaN              NaN     B  
5          120.0          120.0           80.0            560.0     B  
6            NaN            NaN            NaN              NaN     A  
7            NaN            NaN            NaN              NaN     B  

[8 rows x 29 columns]

## Corrigindo erros nas notas

In [17]:
dados1.loc[:, 'NU_NOTA_CN'] /= 10
dados1.loc[:, 'NU_NOTA_CH'] /= 10
dados1.loc[:, 'NU_NOTA_LC'] /= 10
dados1.loc[:, 'NU_NOTA_MT'] /= 10

In [18]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0     1,90E+11            Santa Isabel        22       M                1   
1     1,90E+11             Pinhalzinho        19       M                1   
2     1,90E+11               São Paulo        17       F                1   
3     1,90E+11   São Bernardo do Campo        19       M                1   
4     1,90E+11    São Joaquim da Barra        37       M                2   

   TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ESCOLA  IN_TREINEIRO  \
0            3                 1                1          1             0   
1            3                 1                1          1             0   
2            3                 1                2          3             0   
3            3                 1                2          2             0   
4            1                 1                1          1             0   

   ...  NU_NOTA_MT TP_LINGUA  TP_STATUS_REDACAO  NU_NOTA_COMP1  NU_NOTA_COMP2  \
0  ...       600.2         0                1.0          160.0          200.0   
1  ...         NaN         1                NaN            NaN            NaN   
2  ...       731.8         0                1.0          160.0          120.0   
3  ...       393.2         1                1.0           80.0          160.0   
4  ...         NaN         1                NaN            NaN            NaN   

   NU_NOTA_COMP3  NU_NOTA_COMP4  NU_NOTA_COMP5  NU_NOTA_REDACAO  Q025  
0          180.0          200.0          200.0            940.0     B  
1            NaN            NaN            NaN              NaN     A  
2          200.0          200.0          200.0            880.0     A  
3          120.0          100.0          100.0            560.0     B  
4            NaN            NaN            NaN              NaN     B  

[5 rows x 29 columns]

## Renomear variáveis

In [19]:
dados1 = dados1.rename(columns={'NU_NOTA_REDACAO': 'NOTA_REDACAO', 'NU_NOTA_CN': 'NOTA_CN','NU_NOTA_CH': 'NOTA_CH',
                                'NU_NOTA_LC': 'NOTA_LC','NU_NOTA_MT': 'NOTA_MT'})

dados1 = dados1.rename(columns={'NU_NOTA_COMP1': 'COMP1','NU_NOTA_COMP2': 'COMP2',
                                'NU_NOTA_COMP3': 'COMP3','NU_NOTA_COMP4': 'COMP4',
                                'NU_NOTA_COMP5':'COMP5'})

dados1 = dados1.rename(columns={'NU_IDADE': 'IDADE','TP_SEXO': 'SEXO',
                                'TP_COR_RACA': 'RACA','Q025': 'INTERNET',
                                'TP_ESCOLA':'ESCOLA'})


## Renomeando registros (linhas)

In [22]:
dados1['RACA'] = dados1['RACA'].replace({0:'nao_declarado', 1:'branca', 2:'preta', 3:'parda', 4:'amarela', 5:'indigena'})

In [23]:
dados1["TP_LINGUA"] = dados1["TP_LINGUA"].replace({0:"Inglês", 1:"Espanhol"})

In [24]:
dados1["ESCOLA"] = dados1["ESCOLA"].replace({1:"não_respondeu", 2:"pública", 3:"privada", 4:"exterior"})

In [25]:
dados1["INTERNET"] = dados1["INTERNET"].replace({"A":"Não", "B":"Sim"})

In [26]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO  TP_ESTADO_CIVIL    RACA  \
0     1,90E+11            Santa Isabel     22    M                1   parda   
1     1,90E+11             Pinhalzinho     19    M                1   parda   
2     1,90E+11               São Paulo     17    F                1   parda   
3     1,90E+11   São Bernardo do Campo     19    M                1   parda   
4     1,90E+11    São Joaquim da Barra     37    M                2  branca   

   TP_NACIONALIDADE  TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO  ...  \
0                 1                1  não_respondeu             0  ...   
1                 1                1  não_respondeu             0  ...   
2                 1                2        privada             0  ...   
3                 1                2        pública             0  ...   
4                 1                1  não_respondeu             0  ...   

   NOTA_MT TP_LINGUA  TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  COMP5  \
0    600.2    Inglês                1.0  160.0  200.0  180.0  200.0  200.0   
1      NaN  Espanhol                NaN    NaN    NaN    NaN    NaN    NaN   
2    731.8    Inglês                1.0  160.0  120.0  200.0  200.0  200.0   
3    393.2  Espanhol                1.0   80.0  160.0  120.0  100.0  100.0   
4      NaN  Espanhol                NaN    NaN    NaN    NaN    NaN    NaN   

   NOTA_REDACAO  INTERNET  
0         940.0       Sim  
1           NaN       Não  
2         880.0       Não  
3         560.0       Sim  
4           NaN       Sim  

[5 rows x 29 columns]

# Análise Exploratória e Transformação nos Dados

In [27]:
dados1['IDADE'].value_counts()

IDADE
17    219313
18    147548
19     81236
16     59316
20     53009
       ...  
86         1
4          1
3          1
91         1
2          1
Name: count, Length: 80, dtype: int64

In [28]:
dados1['IDADE'].value_counts().sort_index()

IDADE
2     1
3     1
4     1
5     2
6     2
     ..
80    2
81    3
82    2
86    1
91    1
Name: count, Length: 80, dtype: int64

In [29]:
menores12 = dados1.query('IDADE < 12')['NO_MUNICIPIO_RESIDENCIA'].value_counts()

In [30]:
menores12

NO_MUNICIPIO_RESIDENCIA
São Paulo                 5
Tambaú                    1
Ribeirão Preto            1
Bálsamo                   1
São Bernardo do Campo     1
Embu das Artes            1
Queiroz                   1
Santana de Parnaíba       1
Vargem Grande Paulista    1
Auriflama                 1
Araçatuba                 1
Campinas                  1
Name: count, dtype: int64

In [31]:
dados_maiores11 = dados1.loc[dados1.IDADE > 11]

In [32]:
dados_maiores11['IDADE'].value_counts().sort_index()

IDADE
12        6
13       29
14      355
15    11583
16    59316
      ...  
80        2
81        3
82        2
86        1
91        1
Name: count, Length: 73, dtype: int64

## Treineiros

---
Treinando para o Enem


In [35]:
dados_maiores11.query('IN_TREINEIRO == 1')['IN_TREINEIRO'].value_counts()

IN_TREINEIRO
1    93988
Name: count, dtype: int64

In [36]:
treineiros = dados_maiores11.loc[dados_maiores11.IN_TREINEIRO == 1]

In [37]:
treineiros.to_csv('treineiros_enem_2019_sp.csv', sep=';', encoding='iso-8859-1', index=False)

## Vestibulandos

In [38]:
dados_maiores11.query('IN_TREINEIRO == 0')['IN_TREINEIRO'].value_counts()

IN_TREINEIRO
0    719768
Name: count, dtype: int64

In [39]:
vestibulandos = dados_maiores11.loc[dados_maiores11.IN_TREINEIRO == 0]

In [40]:
vestibulandos.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO  TP_ESTADO_CIVIL    RACA  \
0     1,90E+11            Santa Isabel     22    M                1   parda   
1     1,90E+11             Pinhalzinho     19    M                1   parda   
2     1,90E+11               São Paulo     17    F                1   parda   
3     1,90E+11   São Bernardo do Campo     19    M                1   parda   
4     1,90E+11    São Joaquim da Barra     37    M                2  branca   

   TP_NACIONALIDADE  TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO  ...  \
0                 1                1  não_respondeu             0  ...   
1                 1                1  não_respondeu             0  ...   
2                 1                2        privada             0  ...   
3                 1                2        pública             0  ...   
4                 1                1  não_respondeu             0  ...   

   NOTA_MT TP_LINGUA  TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  COMP5  \
0    600.2    Inglês                1.0  160.0  200.0  180.0  200.0  200.0   
1      NaN  Espanhol                NaN    NaN    NaN    NaN    NaN    NaN   
2    731.8    Inglês                1.0  160.0  120.0  200.0  200.0  200.0   
3    393.2  Espanhol                1.0   80.0  160.0  120.0  100.0  100.0   
4      NaN  Espanhol                NaN    NaN    NaN    NaN    NaN    NaN   

   NOTA_REDACAO  INTERNET  
0         940.0       Sim  
1           NaN       Não  
2         880.0       Não  
3         560.0       Sim  
4           NaN       Sim  

[5 rows x 29 columns]

In [41]:
vestibulandos.shape

(719768, 29)

# Análise das presenças

In [42]:
vestibulandos['TP_PRESENCA_CH'].value_counts()

TP_PRESENCA_CH
1    533142
0    185969
2       657
Name: count, dtype: int64

In [43]:
vestibulandos['TP_PRESENCA_CN'].value_counts()

TP_PRESENCA_CN
1    500805
0    218693
2       270
Name: count, dtype: int64

In [44]:
vestibulandos['TP_PRESENCA_LC'].value_counts()

TP_PRESENCA_LC
1    533142
0    185969
2       657
Name: count, dtype: int64

In [45]:
vestibulandos['TP_PRESENCA_MT'].value_counts()

TP_PRESENCA_MT
1    500805
0    218693
2       270
Name: count, dtype: int64


#### RESUMO DA PRESENÇA NAS PROVAS:

* Ciências da Natureza: 500805 presentes, 218693 faltaram e 270 eliminados.

* Ciências Humanas: 533142 presentes, 185969 faltaram e 657 eliminados.

* Linguagens e Códigos: 533142 presentes, 185969 faltaram e 657 eliminados.

* Matemática: 500805 presentes, 218693 faltaram e 270 eliminados.


In [48]:
vestibulandos['TP_STATUS_REDACAO'].value_counts().sort_index()

TP_STATUS_REDACAO
1.0    519763
2.0       913
3.0      1417
4.0      5513
6.0      3626
7.0       543
8.0       828
9.0       539
Name: count, dtype: int64

## Valores ausentes

In [50]:
vestibulandos.isnull().sum().sort_index()

COMP1                      186626
COMP2                      186626
COMP3                      186626
COMP4                      186626
COMP5                      186626
CO_ESCOLA                  500483
ESCOLA                          0
IDADE                           0
INTERNET                        0
IN_TREINEIRO                    0
NOTA_CH                    186626
NOTA_CN                    218963
NOTA_LC                    186626
NOTA_MT                    218963
NOTA_REDACAO               186626
NO_MUNICIPIO_ESC           500483
NO_MUNICIPIO_RESIDENCIA         0
NU_INSCRICAO                    0
RACA                            0
SEXO                            0
TP_ESTADO_CIVIL                 0
TP_LINGUA                       0
TP_NACIONALIDADE                0
TP_PRESENCA_CH                  0
TP_PRESENCA_CN                  0
TP_PRESENCA_LC                  0
TP_PRESENCA_MT                  0
TP_STATUS_REDACAO          186626
TP_ST_CONCLUSAO                 0
dtype: int64

In [51]:
provas = ['NOTA_REDACAO','NOTA_CN','NOTA_MT','NOTA_LC','NOTA_CH']

In [52]:
vestibulandos[provas].isnull().sum()

NOTA_REDACAO    186626
NOTA_CN         218963
NOTA_MT         218963
NOTA_LC         186626
NOTA_CH         186626
dtype: int64

In [53]:
vestibulandos_presentes = vestibulandos.loc[vestibulandos.TP_PRESENCA_LC == 1]

In [54]:
vestibulandos_presentes = vestibulandos_presentes.loc[vestibulandos_presentes.TP_PRESENCA_CN == 1]

In [56]:
vestibulandos_presentes['TP_PRESENCA_CH'].value_counts()

TP_PRESENCA_MT
1    499940
Name: count, dtype: int64

In [57]:
vestibulandos_presentes['TP_PRESENCA_CN'].value_counts()

TP_PRESENCA_CN
1    499940
Name: count, dtype: int64

In [59]:
vestibulandos_presentes['TP_PRESENCA_LC'].value_counts()

TP_PRESENCA_LC
1    499940
Name: count, dtype: int64

In [60]:
vestibulandos_presentes['TP_PRESENCA_MT'].value_counts()

TP_PRESENCA_MT
1    499940
Name: count, dtype: int64

In [65]:
vestibulandos_presentes[provas].isnull().sum()

NOTA_REDACAO    0
NOTA_CN         0
NOTA_MT         0
NOTA_LC         0
NOTA_CH         0
dtype: int64

## Zeros na Redação

In [66]:
vestibulandos_presentes.query('NOTA_REDACAO == 0')['NOTA_REDACAO'].value_counts()

NOTA_REDACAO
0.0    9181
Name: count, dtype: int64

In [67]:
vestibulandos_presentes.query('COMP1 == 0')['COMP1'].value_counts()

COMP1
0.0    9188
Name: count, dtype: int64

In [68]:
vestibulandos_presentes.query('COMP2 == 0')['COMP2'].value_counts()

COMP2
0.0    9181
Name: count, dtype: int64

In [69]:
vestibulandos_presentes.query('COMP3 == 0')['COMP3'].value_counts()

COMP3
0.0    9197
Name: count, dtype: int64

In [70]:
vestibulandos_presentes.query('COMP4 == 0')['COMP4'].value_counts()

COMP4
0.0    9194
Name: count, dtype: int64

In [72]:
# O valor é mais alto do que o normal, porque esse é o componente mais difícil de aprovar na redação
vestibulandos_presentes.query('COMP5 == 0')['COMP5'].value_counts()

COMP5
0.0    73330
Name: count, dtype: int64

#### NOTAS ZEROS EM REDAÇÃO:

* REDAÇÃO (final): 9181 notas zeros devido a não cumprimento das regras ou tirou zero em todos os cinco componentes.

* COMPONENTE 1(Demonstrar domínio da modalidade escrita formal da Língua Portuguesa.): 9188 notas zeros somente neste item.

* COMPONENTE 2 (Compreender a proposta de redação e aplicar conceitos das várias áreas de conhecimento para desenvolver o tema, dentro dos limites estruturais do texto dissertativo-argumentativo em prosa): 9181.

* COMPONENTE 3 (Selecionar, relacionar, organizar e interpretar informações, fatos, opiniões e argumentos em defesa de um ponto de vista): 9197 notas zeros somente neste item.

* COMPONENTE 4 (Demonstrar conhecimento dos mecanismos linguísticos necessários para a construção da argumentação.): 9194 notas zeros somente neste item.

* COMPONENTE 5 (Elaborar proposta de intervenção para o problema abordado, respeitando os direitos humanos): 73330 notas zeros somente neste item.


In [73]:
vestibulandos_presentes.query('NOTA_REDACAO == 0')['NOTA_REDACAO'].value_counts()

NOTA_REDACAO
0.0    9181
Name: count, dtype: int64

In [74]:
vestibulandos_presentes.query('NOTA_CN == 0')['NOTA_CN'].value_counts()

NOTA_CN
0.0    36
Name: count, dtype: int64

In [75]:
vestibulandos_presentes.query('NOTA_CH == 0')['NOTA_CH'].value_counts()

NOTA_CH
0.0    343
Name: count, dtype: int64

In [77]:
vestibulandos_presentes.query('NOTA_LC == 0')['NOTA_LC'].value_counts()

NOTA_LC
0.0    135
Name: count, dtype: int64

In [78]:
vestibulandos_presentes.query('NOTA_MT == 0')['NOTA_MT'].value_counts()

NOTA_MT
0.0    56
Name: count, dtype: int64

#### NOTAS ZEROS DOS INSCRITOS CONCORRENTES AO VESTIBULAR:

* Matemática = 56

* Ciências da Natureza = 36

* Linguagens e Códigos = 135

* Ciências Humanas = 343

* Redação = 9181

In [79]:
vestibulandos_presentes.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO  TP_ESTADO_CIVIL    RACA  \
0     1,90E+11            Santa Isabel     22    M                1   parda   
2     1,90E+11               São Paulo     17    F                1   parda   
3     1,90E+11   São Bernardo do Campo     19    M                1   parda   
5     1,90E+11              Embu-Guaçu     17    F                1   parda   
8     1,90E+11               Guarulhos     23    F                1  branca   

   TP_NACIONALIDADE  TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO  ...  \
0                 1                1  não_respondeu             0  ...   
2                 1                2        privada             0  ...   
3                 1                2        pública             0  ...   
5                 1                2        pública             0  ...   
8                 1                1  não_respondeu             0  ...   

   NOTA_MT TP_LINGUA  TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  COMP5  \
0    600.2    Inglês                1.0  160.0  200.0  180.0  200.0  200.0   
2    731.8    Inglês                1.0  160.0  120.0  200.0  200.0  200.0   
3    393.2  Espanhol                1.0   80.0  160.0  120.0  100.0  100.0   
5    419.2    Inglês                1.0  120.0  120.0  120.0  120.0   80.0   
8    796.1    Inglês                1.0  160.0  200.0  180.0  180.0  160.0   

   NOTA_REDACAO  INTERNET  
0         940.0       Sim  
2         880.0       Não  
3         560.0       Sim  
5         560.0       Sim  
8         880.0       Sim  

[5 rows x 29 columns]

In [84]:
vestibulandos_presentes.query('TP_NACIONALIDADE == 1')['TP_NACIONALIDADE'].value_counts()

TP_NACIONALIDADE
1    487784
Name: count, dtype: int64

In [85]:
vestibulandos_presentes = vestibulandos_presentes.drop(columns=['TP_NACIONALIDADE', 'TP_ESTADO_CIVIL', 'IN_TREINEIRO', ], )

In [87]:
vestibulandos_presentes.head(2)

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO   RACA  TP_ST_CONCLUSAO  \
0     1,90E+11            Santa Isabel     22    M  parda                1   
2     1,90E+11               São Paulo     17    F  parda                2   

          ESCOLA   CO_ESCOLA NO_MUNICIPIO_ESC  TP_PRESENCA_CN  ...  NOTA_MT  \
0  não_respondeu         NaN              NaN               1  ...    600.2   
2        privada  15567761.0            Belém               1  ...    731.8   

   TP_LINGUA  TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  COMP5  \
0     Inglês                1.0  160.0  200.0  180.0  200.0  200.0   
2     Inglês                1.0  160.0  120.0  200.0  200.0  200.0   

   NOTA_REDACAO  INTERNET  
0         940.0       Sim  
2         880.0       Não  

[2 rows x 26 columns]

In [88]:
vestibulandos_presentes.shape

(499940, 26)

# Salvando dados tratados

In [89]:
vestibulandos_presentes.to_csv('enem_2019_tratados.csv', sep=';', encoding='iso-8859-1', index=False)